File: pfm-ui/src/main/java/com/tink/pfmui/configuration/TelephonyUtils.java
package com.tink.pfmui.configuration;

import android.content.Context;
import android.telephony.TelephonyManager;
import androidx.annotation.Nullable;

final class TelephonyUtils {

	@Nullable
	public static String getTelephonySimMarket(Context context) {
		TelephonyManager telephonyManager = (TelephonyManager) context
			.getSystemService(Context.TELEPHONY_SERVICE);

		if (telephonyManager != null) {
			return telephonyManager.getSimCountryIso().toUpperCase();
		} else {
			return null;
		}
	}
}

File: moneymanager-ui/src/main/java/com/tink/moneymanagerui/configuration/TelephonyUtils.kt
package com.tink.moneymanagerui.configuration

import android.content.Context
import android.telephony.TelephonyManager
import java.util.Locale

internal object TelephonyUtils {
    fun getTelephonySimMarket(context: Context): String {
        val telephonyManager = context
            .getSystemService(Context.TELEPHONY_SERVICE) as TelephonyManager
        return telephonyManager.simCountryIso.uppercase(Locale.getDefault())
    }
}
